pkg/scaling/scale_handler.go

```diff
@@ -296,15 +296,21 @@ func min(x, y int64) int64 {
 func (h *scaleHandler) buildScalers(withTriggers *kedav1alpha1.WithTriggers, podTemplateSpec *corev1.PodTemplateSpec, containerName string) ([]scalers.Scaler, error) {
 	logger := h.logger.WithValues("type", withTriggers.Kind, "namespace", withTriggers.Namespace, "name", withTriggers.Name)
 	var scalersRes []scalers.Scaler
-
-	resolvedEnv, err := resolver.ResolveContainerEnv(h.client, logger, &podTemplateSpec.Spec, containerName, withTriggers.Namespace)
-	if err != nil {
-		return scalersRes, fmt.Errorf("error resolving secrets for ScaleTarget: %s", err)
+	var err error
+	resolvedEnv := make(map[string]string)
+	if podTemplateSpec != nil {
+		resolvedEnv, err = resolver.ResolveContainerEnv(h.client, logger, &podTemplateSpec.Spec, containerName, withTriggers.Namespace)
+		if err != nil {
+			return scalersRes, fmt.Errorf("error resolving secrets for ScaleTarget: %s", err)
+		}
 	}
 
 	for i, trigger := range withTriggers.Spec.Triggers {
-		authParams, podIdentity := resolver.ResolveAuthRef(h.client, logger, trigger.AuthenticationRef, &podTemplateSpec.Spec, withTriggers.Namespace)
-
+		var authParams map[string]string
+		var podIdentity string
+		if podTemplateSpec != nil {
+			authParams, podIdentity = resolver.ResolveAuthRef(h.client, logger, trigger.AuthenticationRef, &podTemplateSpec.Spec, withTriggers.Namespace)
```

Comment:
We shouldn't skip the whole `resolver.ResolveAuthRef()` call, if you look inside, it is resolving env vars from TriggerAuthentication resource as well (which should be ok even if you don't have podSpec).

Reply:
I see, I added it back. but within `resolver.ResolveAuthRef()` the calll to `ResolveContainerEnv` assume `podSpec` is not `nil` thus I added a condition to it

Reply:
@zroubalik let me know whether there are more changes I should add